File: trunk/src/core/src/main/java/org/semachina/jena/datatype/factory/MonthFactory.java
package org.semachina.jena.datatype.factory;

import com.hp.hpl.jena.vocabulary.XSD;
import org.semachina.jena.datatype.types.Month;
import org.semachina.jena.impl.SimpleRDFDatatypeImpl;

public class MonthFactory extends SimpleRDFDatatypeImpl<Month> {
    public MonthFactory() {
        super(XSD.gMonth.getURI(), Month.class);
    }

    @Override
    public Month parseLexicalForm(String lexicalForm) throws Exception {
        return new Month(lexicalForm);
    }

    @Override
    public String toLexicalForm(Month cast) throws Exception {
        return cast.toString();
    }

    @Override
    public Object cannonicalise(Object value) {
        if (value instanceof Number) {
            return new Month(((Number) value).intValue());
        }
        return super.cannonicalise(value);
    }
}

File: trunk/src/core/src/main/scala/org/semachina/jena/datatype/factory/MonthFactory.scala
package org.semachina.jena.datatype.factory

import com.hp.hpl.jena.vocabulary.XSD
import org.semachina.jena.datatype.SimpleRDFDatatype
import org.semachina.jena.datatype.types.Month

class MonthFactory extends SimpleRDFDatatype[Month] (
  XSD.gMonth.getURI,
  { lexicalForm: String => new Month(lexicalForm) }) {

  override def cannonicalise(value: AnyRef): Month = {
    if (value.isInstanceOf[Number]) {
      return new Month((value.asInstanceOf[Number]).intValue)
    }
    return super.cannonicalise(value)
  }
}